packages/mobile/src/send/SendAmount.tsx

```diff
@@ -1,254 +1,217 @@
   convertLocalAmountToDollars,
 } from 'src/localCurrency/convert'
-import { getLocalCurrencyCode, getLocalCurrencyExchangeRate } from 'src/localCurrency/selectors'
-import { HeaderTitleWithBalance, headerWithBackButton } from 'src/navigator/Headers'
+import {
+  getLocalCurrencyCode,
+  getLocalCurrencyExchangeRate,
+  getLocalCurrencySymbol,
+} from 'src/localCurrency/selectors'
 import { navigate } from 'src/navigator/NavigationService'
 import { Screens } from 'src/navigator/Screens'
 import { StackParamList } from 'src/navigator/types'
 import { getRecipientVerificationStatus, Recipient } from 'src/recipients/recipient'
-import { RootState } from 'src/redux/reducers'
-import { PaymentInfo } from 'src/send/reducers'
+import useSelector from 'src/redux/useSelector'
 import { getRecentPayments } from 'src/send/selectors'
-import { dailyAmountRemaining, getFeeType } from 'src/send/utils'
+import { dailyAmountRemaining, getFeeType, isPaymentLimitReached } from 'src/send/utils'
 import DisconnectBanner from 'src/shared/DisconnectBanner'
 import { fetchDollarBalance } from 'src/stableToken/actions'
-import { withDecimalSeparator } from 'src/utils/withDecimalSeparator'
+import { stableTokenBalanceSelector } from 'src/stableToken/reducer'
 
 export interface TransactionDataInput {
   recipient: Recipient
   amount: BigNumber
-  reason: string
   type: TokenTransactionType
+  reason: string
   firebasePendingRequestUid?: string | null
 }
 
-const AmountInput = withDecimalSeparator(
-  withTextInputLabeling<ValidatedTextInputProps<DecimalValidatorProps>>(ValidatedTextInput)
-)
-const CommentInput = withTextInputLabeling<TextInputProps>(TextInput)
+type RouteProps = StackScreenProps<StackParamList, Screens.SendAmount>
+type Props = RouteProps
 
-interface State {
-  amount: string
-  reason: string
-}
+const { decimalSeparator } = getNumberFormatSettings()
 
-type OwnProps = StackScreenProps<StackParamList, Screens.SendAmount>
-type Props = StateProps & DispatchProps & OwnProps & WithTranslation
+function SendAmount(props: Props) {
+  const dispatch = useDispatch()
 
-interface StateProps {
-  dollarBalance: string
-  estimateFeeDollars: BigNumber | undefined
-  defaultCountryCode: string
-  feeType: FeeType | null
-  localCurrencyCode: LocalCurrencyCode
-  localCurrencyExchangeRate: string | null | undefined
-  recipient: Recipient
-  recipientVerificationStatus: RecipientVerificationStatus
-  addressValidationType: AddressValidationType
-  recentPayments: PaymentInfo[]
-}
+  const isRequest = props.route.params?.isRequest ?? false
+  const recipient = props.route.params?.recipient
 
-interface DispatchProps {
-  fetchDollarBalance: typeof fetchDollarBalance
-  showMessage: typeof showMessage
-  showError: typeof showError
-  hideAlert: typeof hideAlert
-  fetchAddressesAndValidate: typeof fetchAddressesAndValidate
-}
+  React.useEffect(() => {
+    dispatch(fetchDollarBalance())
+    if (recipient.e164PhoneNumber) {
```

Comment:
I believe there's a bug here, possibly from @tarikbellamine's changes for SecureSend.
We used to check if the recipient type was address or qrcode and skip the lookup in those cases, but that check seems to be gone:
https://github.com/celo-org/celo-monorepo/blob/6e6a9e5105796286064c4c7a9f903137d7d674d3/packages/mobile/src/send/SendAmount.tsx#L163

Reply:
@tarikbellamine do we handle this case somewhere else now?

Reply:
This was intentional though potentially incorrect. My overarching reasoning was that we should fetch the latest mapping whenever we can as it's better to be safe that sorry with SecureSend.

The false positives will at most require 2 or 3 more clicks and an extra 10-20 seconds to go through SecureSend as they already have their QR code or address ready to verify. Also worth noting that recipients caught here will have to go through SecureSend eventually anyways, we are just pulling the process forward to a transaction that may have not needed it.

A false negative is a loss of funds. Admittedly, I don't know enough about how we assign recipient types to say how likely I think a false negative would be.

Given the uncertainty a relatively low cost of playing it safe, that's the direction it went in. What do you think I could be missing?

Reply:
That's all reasonable but we may still want to skip for these cases.
In the `address` case, I don't think we'll have e164Number set so we're probably fine either way.
In the QRCode case, my preference is that QR codes *always* work seamlessly. If we force a lookup in that case, we're vulnerable to a whole bunch of possible issues like:
1. PGPNP being down/unavailable
2. User being out of PGPNP lookups and having to buy more first
3. Recipient has incorrect/outdated data on-chain

QR code should take precedence over the on-chain mappings IMO because the user know who they're interacting with.

Reply:
I see. I wasn't aware of the costs associated with doing a lookup and agree that it is preferable to have one method (QR scanning) that always works, no matter what.

In that case, I am good with reverting back to the old conditional logic. From a process perspective, does it make more sense for me to open up a quick PR or let @i1skn revert that one line?

Reply:
or just propose a change right in this PR using 

<img width="496" alt="Screenshot 2020-05-27 at 20 19 23" src="https://user-images.githubusercontent.com/6160124/83057613-755f4c00-a057-11ea-9637-5a611d2ad751.png">

Reply:
Thanks for the heads up on that feature. Looking at it a bit more, I think it may make more sense to submit a quick PR because it requires importing `Recipient` type and I'd feel better if I could also run the tests with the augmented logic to make sure there aren't downstream issues.

Reply:
FYI it's been [merged into master](https://github.com/celo-org/celo-monorepo/pull/3915)